services/matching-service/domain/match_validators.ts

```diff
@@ -0,0 +1,44 @@
+// Initialize queues within matching-socketio service
+import { Match, ids } from "../start";
+
+export function extractDifficulty(difficulty: string) {
+    const DIFFICULTIES = [
```

Comment:
Im not sure how using enums improves the code? I am not having to call e.g. Difficulty.EASY anywhere in the code. Also, with enums, to parse and check whether its a valid difficulty/category, I would still have to iterate through every enum member (same as an array).

Reply:
if you have them as enums instead of constants, you can directly parse them as enums and not need any validators, and error checking is done for you so you don't have to check for undefined in your start.ts. My concern is more of unnecessary code and not really a concern about performance